ls1intum/Artemis
# Bugfix/drag and drop quiz drop zones

### Checklist

### Motivation and Context
The DropLocations in "Drag and Drop" quizzes are not positioned correctly. There were two issues:
1. The DropLocations where calculated wrongly, because the UML-Diagram model size was used instead of the generated SVG size
2. The positions of DropLocations where rounded, thereby introducing a small error

### Description
1. Change DropLocations data types from Integer to Double and round relative positions to second decimal, thereby reducing rounding errors
2. Use svg bounds to calculate relative position -> model size is the full model not cropped to the bounding box of the svg, which led to wrong positions of DropLocations on background image

### Steps for Testing

1. Log in to Artemis
2. Navigate to "Course Administration"
3. Create a Drag and Drop quiz
4. Use any kind of diagram type
5. mark elements as interactive
6. Generate diagram in course
7. Navigate to course exercises
8. Preview generated diagram
9. Check locations of bounding boxes

### Screenshots
Before
![drag_and_drop](https://user-images.githubusercontent.com/22955066/79164701-e1f0f500-7de1-11ea-9c1d-9021954bfc4f.png)
After
![screenshot_2](https://user-images.githubusercontent.com/22955066/79164707-e4ebe580-7de1-11ea-8335-ca22af3731f9.png)
